### PR TITLE
Remove producer dependency

### PR DESCRIPTION
**Pull request by CarstenWalther**
The consumer should not need to know about the producer.

I removed the product popularity feature and the product selection based on these popularities. Instead, the buying behaviors get all offers of all product types to choose from.

The api route `update_product_details` is no longer needed and is removed.

I removed the sigmoid buying behavior because it used product prices from the producer.
I tested all other buying behaviors and they worked. We can close issue #8.

### CODE REVIEW

**Comment by Bouncner** [comment]
What does "tested all other buying behaviors and they worked" mean exactly?

**Comment by CarstenWalther** [comment]
I ran each buying behavior and couldn't reproduce the crash.

**Comment by Bouncner** [comment]
Ok ... I hoped for "I wrote some tests and they all pass". ;)

File: README.md [hunk 1]
```diff
@@ -28,23 +28,18 @@ The consumer is written in Rails. Ensure to have the following components instal
...
 
-Currently, two ENV params are needed:
-* PRICEWARS_PRODUCER_URL
-* PRICEWARS_MARKETPLACE_URL
-
-If they are not set, the deployed VM urls are automatically used.
-
-One may simple export them to the related system:
+The marketplace url is configured in the environment variable `PRICEWARS_MARKETPLACE_URL`.
+If it is not set, a default url to a deployed VM is used.
```
Comment (Bouncner) [inline README.md]: The default is the docker-internal path (`merchant`) to the marketplace, right?
Or what deployed VM do you mean?

File: README.md [hunk 2]
```diff
@@ -53,7 +48,7 @@ The consumer is defined via behaviors which are implemented in *lib/buyingbehavi
 
 ### Consumer Behaviors
 
-Via settings, the distribution across those available behaviors is defined as percentage. In the consumer logic then, each behavior is [called and executed](https://github.com/hpi-epic/pricewars-consumer/blob/master/app/controllers/setting_controller.rb#L127) dynamically based on the provided behavior method name and its distribution.
+Via settings, the distribution across those available behaviors is defined as percentage. In the consumer logic then, each behavior is called and executed dynamically based on the provided behavior method name and its distribution.
```
Comment (Bouncner) [inline README.md]: Which settings?

File: README.md (hunk 1 above)
Comment (CarstenWalther) [reply README.md]: The default url was `vm-mpws2016hp1-04.eaalab.hpi.uni-potsdam.de:8080/marketplace`. I changed that to `http://marketplace:8080`.

File: README.md (hunk 2 above)
Comment (CarstenWalther) [reply README.md]: I formulated this more clearly

**Comment by CarstenWalther** [comment]
I updated the consumer submodule.

**Pull request merged**